### ISSUE TITLE
Older version with dash shows as newer

### ISSUE THREAD

**Issue by brewin** [jmfayard/refreshVersions#223]
refreshVersions incorrectly shows 'x.y-whatever' as newer than 'x.y.z'

```
version.com.russhwolf..multiplatform-settings-no-arg=0.6.1
##                                       # available=0.6-1.4-M1
##                                       # available=0.6-1.4-M2
##                                       # available=0.6-1.4-M3
##                                       # available=0.6-1.4.0-rc
```

**Comment by jmfayard** [comment]
@brewin developers are crazy creative with their naming scheme

The simplest solution is to use a `rejectVersionIf`
https://jmfayard.github.io/refreshVersions/update-dependencies/#filter-which-versions-are-added-to-versionsproperties

**Comment by LouisCAD** [comment]
We can still tweak the version ordering system some day, because the scheme described here is quite common.

I might do it sometime next year as a mental exercise.

**Comment by rlatapy-luna** [comment]
Is there any update on this? I currently reject snapshots with same version as release using `candidate.value.startsWith(current.value) && candidate.stabilityLevel == StabilityLevel.Snapshot`, but a custom sorting would be better.

**Comment by LouisCAD** [comment]
That's expected. A stable release supersedes the snapshots for the same version.

If you are on 5.0.0, it's newer than 5.0.0-SNAPSHOT, but lower than 5.0.1-SNAPSHOT or anything beyond.

**Comment by LouisCAD** [comment]
No changes to expect on that in the foreseeable future, it's low priority, I already tuned the sorting algorithm to make it better, and you have a workaround.

**Comment by LouisCAD** [comment]
Also, snapshots for the same release should not be shown.

**Comment by rlatapy-luna** [comment]
> Also, snapshots for the same release should not be shown.

Yes but in our case we bump the version immediately after releasing a new version. So our snapshots during feature dev are like `1.5.0-my_feature_under_dev-SNAPSHOT` and won't be superseded by `1.5.0` after release.
I agree that this may not be the main case, but an api to inject a kind of custom comparator could be nice.
